lumen/ai/agents.py

```diff
@@ -157,14 +157,35 @@ def _get_schema(self, source: Source | Pipeline, table: str = None):
         schema = dict(schema)
         for field, spec in schema.items():
             if "enum" in spec:
-                schema[field] = dict(spec, enum=spec["enum"][:5])
+                spec.pop("enum")
+            if "inclusiveMinimum" in spec:
+                spec["min"] = spec.pop("inclusiveMinimum")
+            if "inclusiveMaximum" in spec:
+                spec["max"] = spec.pop("inclusiveMaximum")
         return schema
 
+
+    async def _get_closest_tables(self, messages: list | str, tables: list[str], n: int = 3) -> list[str]:
+        system = (
+            "You are great at extracting keywords based on the user query to find the correct table."
+        )
+        table_keywords = (await self.llm.invoke(
+            messages,
+            system=system,
+            response_model=FuzzyTable,
+            allow_partial=False
+        )).keywords
+        tables = tuple(table.replace('"', "") for table in tables)
+        closest_tables = difflib.get_close_matches(table_keywords, tables, n=n)
+        if not closest_tables:
+            closest_tables = tables[:n]
```

Comment:
I think here is a great opportunity to introduce wait-for-human-input-from-widget.